onyo/commands/mkdir.py

```diff
@@ -22,7 +23,19 @@ def mkdir(args, opdir):
 
     try:
         repo.mkdir(args.directory)
-        repo.commit(repo.generate_commit_message(message=args.message,
-                                                 cmd="mkdir"))
     except (FileExistsError, OnyoProtectedPathError):
         sys.exit(1)
+
+    # commit changes
+    staged = sorted(repo.files_staged)
+    if staged:
```

Comment:
I think you are right. Though it is always like that, in case something goes wrong in the mkdir (or other commands) with the "git add" so that files are not staged, but changed.

And alternatively, when a command would not change anything, but miss to error, this would also stop the code afterwards.

But I think because of argparse and all the other checks in functions, this technically could be removed.

Reply:
> And alternatively, when a command would not change anything, but miss to error, this would also stop the code afterwards.

Yes. This is a feature to me. And in such a case we do want Onyo to explode.